### PR TITLE
[eclipse/xtext#1952] converting xtend code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1952] converting xtend code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
please review

File: org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeContentProposalProvider.java [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+
+public class XbaseIdeContentProposalProvider extends IdeContentProposalProvider {
+	public static class ValidFeatureDescription implements Predicate<IEObjectDescription> {
+		@Inject
+		private OperatorMapping operatorMapping;
+
+		@Override
+		public boolean apply(IEObjectDescription input) {
+			if (input instanceof IIdentifiableElementDescription) {
+				if (!((IIdentifiableElementDescription) input).isVisible()
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeContentProposalProvider.java:67]: Please use a local veraible for the casted input

File: org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeContentProposalProvider.java [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+				return;
+			}
+		}
+		if (model instanceof XMemberFeatureCall) {
+			ICompositeNode node = NodeModelUtils.getNode(model);
+			if (isInMemberFeatureCall(model, node.getEndOffset(), context)) {
+				return;
+			}
+		}
+		createLocalVariableAndImplicitProposals(model, IExpressionScope.Anchor.AFTER, context, acceptor);
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeContentProposalProvider.java:235]: Hmm should we use static imports for the Anchor enum literals?

File: org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeContentProposalProvider.java [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+		}
+		return null;
+	}
+
+	protected String _getConcreteSyntaxRuleName(RuleCall ruleCall) {
+		return ruleCall.getRule().getName();
+	}
+
+	@Override
+	public void createProposals(AbstractElement assignment, ContentAssistContext context,
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeContentProposalProvider.java:426]: Parameter name `assignment` is misleading

**Review by szarnekow — APPROVED** [review]

**Pull request merged**